app/controllers/underpayments/UnderpaymentDetailSummaryController.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.underpayments
+
+import controllers.actions.{DataRequiredAction, DataRetrievalAction, IdentifierAction}
+import models.UnderpaymentDetail
+import models.underpayments.UnderpaymentAmount
+import pages.underpayments.{UnderpaymentDetailSummaryPage, UnderpaymentDetailsPage}
+import play.api.i18n.{I18nSupport, Messages}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content.{HtmlContent, Text}
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist._
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.ViewUtils.displayMoney
+import views.html.underpayments.UnderpaymentDetailSummaryView
+
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class UnderpaymentDetailSummaryController @Inject()(identify: IdentifierAction,
```

Comment:
I think this might be better renamed as UnderpaymentDetailsConfirmController as this is the point they are confirming the details they have just put in